app/models/task.ts

```diff
@@ -12,68 +13,59 @@ import { TaskDependencies } from "./task-dependencies";
 import { TaskExecutionInformation } from "./task-execution-information";
 import { TaskExitConditions } from "./task-exit-conditions";
 
-// tslint:disable:variable-name
-const TaskRecord = Record({
-    id: null,
-    displayName: null,
-    url: null,
-    eTag: null,
-    lastModified: null,
-    creationTime: null,
-    state: null,
-    stateTransitionTime: null,
-    previousState: null,
-    previousStateTransitionTime: null,
-    commandLine: null,
-    runElevated: null,
-    exitConditions: new TaskExitConditions(),
-    resourceFiles: [],
-    environmentSettings: null,
-    affinityInfo: null,
-    constraints: null,
-    executionInfo: null,
-    nodeInfo: null,
-    multiInstanceSettings: null,
-    stats: null,
-    dependsOn: null,
-    applicationPackageReferences: null,
-});
-
+export interface TaskAttributes {
+    id: string;
+    displayName: string;
+    url: string;
+    eTag: string;
+    lastModified: Date;
+    creationTime: Date;
+    state: TaskState;
+    stateTransitionTime: Date;
+    previousState: TaskState;
+    previousStateTransitionTime: Date;
+    commandLine: string;
+    runElevated: boolean;
+    exitConditions: TaskExitConditions;
+    resourceFiles: ResourceFile[];
+    environmentSettings: NameValuePair[];
```

Comment:
no this is the input(what is returned by the server) so must be the native javascript object